### PR TITLE
Add Tasks functionality

### LINKED ISSUE #22: As a team leader, I can create tasks for the team
... so that I can track things being done and things that need to be done.

### PR DESCRIPTION
**Pull request by Devanshshah1309**
## Changes
1. Add command `add_task` and corresponding parser.
2. Added `Task` class and `TaskList` class to handle task manipulations.
3. Add basic tests.
4. Add command `list_tasks`.

## Things to Note:
1. Tasks are not yet stored into the JSON file since they're associated with teams and teams are not being stored.
2. When more attributes (e.g. deadlines) are added to `Task`, other methods (e.g. `equals`) will also have to change.
3. This design supports multiple teams since each team has a single `TaskList` object. Suggestion for assigning a person to a task is to just create an additional field in `Task` called `assignee` that tracks if a task has been assigned yet (and to whom if so).
4. The UI is not updated since it too requires the UI for the Team to be updated first (the `TaskCard` and `TaskListPanel` have been created already though).

## When Reviewing
Look out for
1. Edge cases in `add_task` and `list_tasks`
2. Any typos in the text displayed showed to user.
3. (Improving the UI lol)

Requires 2 reviews to merge.

Fixes #22

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListTasksCommand.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import static seedu.address.model.team.TaskList.NO_TASKS;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+
+
+/**
+ * Adds a task to the current team.
+ */
```
Comment (changjy44) [inline src/main/java/seedu/address/logic/commands/ListTasksCommand.java:13]: Duplicate description in ListTaskCommand from AddTaskCommand, do update!

File: src/main/java/seedu/address/logic/commands/ListTasksCommand.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        + ": Lists all the tasks of the current team.\n"
+        + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_ADD_TASK_SUCCESS = "Tasks: \n%1$s";
+
+    /**
+     * Lists all tasks of the current team.
+     */
+    public ListTasksCommand() {
+    }
```
Comment (changjy44) [inline src/main/java/seedu/address/logic/commands/ListTasksCommand.java:27]: Unnecessary javadocs for constructor, might want to shift it up.

File: src/main/java/seedu/address/model/AddressBook.java [hunk 3]
```diff
@@ -17,6 +19,7 @@
 public class AddressBook implements ReadOnlyAddressBook {
 
     private final UniquePersonList persons;
+    private final TaskList taskList;
```
Comment (changjy44) [inline src/main/java/seedu/address/model/AddressBook.java:22]: Not exactly clear why AddressBook needs a taskList reference, should taskList be only encapsulated inside Teams?

File: src/main/java/seedu/address/model/AddressBook.java [hunk 4]
```diff
@@ -56,11 +61,18 @@ public void setPersons(List<Person> persons) {
...
     }
 
+    /**
+     * Replaces the contents of the person list with {@code persons}.
+     * {@code persons} must not contain duplicate persons.
+     */
+    public void setTaskList(List<Task> tasks) {
+        this.taskList.setTasks(tasks);
+    }
+
```
Comment (changjy44) [inline src/main/java/seedu/address/model/AddressBook.java:75]: Javadocs description does not match the method, do update again.

File: src/main/java/seedu/address/model/team/TaskList.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            throw new DuplicatePersonException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
+     * {@code target} must exist in the list.
+     * The person identity of {@code editedPerson} must not be the same as another existing person in the list.
+     */
```
Comment (changjy44) [inline src/main/java/seedu/address/model/team/TaskList.java:50]: Javadocs method description does not match, do update!

**Review by changjy44 — CHANGES_REQUESTED** [review]
Do update the relevant parts that needs to be changed and self merge when done, thanks.

**Comment by Devanshshah1309** [comment]
> Do update the relevant parts that needs to be changed and self merge when done, thanks. EDIT: Do configure your TaskListPanel to appear in the UI as well, I will build the Teams UI around your TaskListPanel.

Not sure if it is right to have the `TaskListPanel` without the `TeamsListPanel` though

**Comment by Devanshshah1309** [comment]
Do you think we can implement all the UI changes once both the `Team` and `Task` implementations are stabilised? We might need to discuss with the team on how we want the UI to look like for different commands (e.g. Do we always want tasks to be shown on the right panel? Or, do we want the tasks to replace the `TeamListPanel` when `list_tasks` is run?). I think it might be better to get the logic + implementation of designs done before we move on to fixing the UI. What do you think? (hopefully if the `Team` and `Tasks` is settled by tuesday, we can still implement the UI by Friday, in time for v1.2)?

**Comment by ling-ui** [comment]
> Do you think we can implement all the UI changes once both the `Team` and `Task` implementations are stabilised? We might need to discuss with the team on how we want the UI to look like for different commands (e.g. Do we always want tasks to be shown on the right panel? Or, do we want the tasks to replace the `TeamListPanel` when `list_tasks` is run?). I think it might be better to get the logic + implementation of designs done before we move on to fixing the UI. What does everyone think? @changjy44 @mjgui @marcuspang @ling-ui (hopefully if the `Team` and `Tasks` is settled by tuesday, we can still implement the UI by Friday, in time for v1.2)?

I suppose we can talk about the UI during our next meeting, perhaps we focus on getting the core features up first.

**Review by changjy44 — APPROVED** [review]
Looks good to merge

File: src/main/java/seedu/address/logic/commands/ListTasksCommand.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ * Lists all tasks of the current team.
+ */
+public class ListTasksCommand extends Command {
+    public static final String COMMAND_WORD = "list_tasks";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Lists all the tasks of the current team.\n"
+        + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_ADD_TASK_SUCCESS = "Tasks: \n%1$s";
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/ListTasksCommand.java:21]: This should be `MESSAGE_LIST_TASK` right?

**Pull request merged**